# **TITANIC SURVIVOR ANALYSIS:**

In [ ]:
import pandas as pd
import numpy as np

**READING DATA:**

> We use pandas read_csv function to read the csv file in python.And then covert it into dataframe using .DataFrame Function. 



In [ ]:
df = pd.DataFrame(pd.read_csv('/content/sample_data/train (1).csv'))
df.head(15)   # We used df.head(10) to specifically view first 15 columns.

PassengerId  Survived  Pclass  ...     Fare Cabin  Embarked
0             1         0       3  ...   7.2500   NaN         S
1             2         1       1  ...  71.2833   C85         C
2             3         1       3  ...   7.9250   NaN         S
3             4         1       1  ...  53.1000  C123         S
4             5         0       3  ...   8.0500   NaN         S
5             6         0       3  ...   8.4583   NaN         Q
6             7         0       1  ...  51.8625   E46         S
7             8         0       3  ...  21.0750   NaN         S
8             9         1       3  ...  11.1333   NaN         S
9            10         1       2  ...  30.0708   NaN         C
10           11         1       3  ...  16.7000    G6         S
11           12         1       1  ...  26.5500  C103         S
12           13         0       3  ...   8.0500   NaN         S
13           14         0       3  ...  31.2750   NaN         S
14           15         0       3  ...   7.8542   NaN         S

[15 rows x 12 columns]

In [ ]:
df.shape   # This will return us the no.of rows and columns in a tuple format.

(891, 12)

# HANDELING THE NULL VALUES:
The dataset may contain some rows and colums in which the values are missing.We cant leave those missing values as it is.
In this case we have 2 options:

> Either drop the entire row or column.


> Fill the missing values with some appropriate value .Like mean of all values of that column.





In [ ]:
df.isnull().sum()

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64

# **Seperating out the columns which have more than 45% of values missing in the dataset:**

> df.isnull().sum() returns a Pandas series with column names as the label index.

> And total count of the null values in a column and its value.


> And your sorting only those columns which have more than 45% of data missing.







In [ ]:
drop_col = df.isnull().sum()[df.isnull().sum()>(45/100 * df.shape[0])]
                                         
drop_col

Cabin    687
dtype: int64

In [ ]:
drop_col.index

Index(['Cabin'], dtype='object')


*   The **Cabin** column has lot of null values and somewhat the given column is not of any use as it has only string values.

*   So now we are droping the **Cabin** cloumn from the dataset by implementing the code below.




In [ ]:
df.drop(drop_col.index , axis=1, inplace=True)
df.isnull().sum()

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Embarked         2
dtype: int64



1.   **df.fillna**  will fill a specific number  in all the columns. 
2.   Now in **Age** colum we are filling the empty places with the mean of the age.



In [ ]:
df.fillna(df.mean(), inplace=True)
df.isnull().sum()

PassengerId    0
Survived       0
Pclass         0
Name           0
Sex            0
Age            0
SibSp          0
Parch          0
Ticket         0
Fare           0
Embarked       2
dtype: int64

The .describe() function will tell all the details of a specific column.

In [ ]:
df['Embarked'].describe()

count     889
unique      3
top         S
freq      644
Name: Embarked, dtype: object

Now we can see that the above code has shown all the details about the colum **Embarked.**


*   Count is telling us the total number of non null values. we can see that thier are total number of 889 non-null values in the column **Embarked**
*   There are total number of 3 unique values in the column  **Embarked**(S,C,Q)


*   S  is the most frequently occurred entry in the column **Embarked** with a frequency of 644.

  




In [ ]:
df['Sex'].describe()

count      891
unique       2
top       male
freq       577
Name: Sex, dtype: object



> we can apply the same  to the **Sex** column. We can see that there are more no.of males in the ship than females.(i.e, no.of males= 577)





>  For **Embarked** attribute, we fill the Null-values with the most frequent value in the column.(i.e, S)


> We are doing this beacause the **Embarked** attribute only has string values.we can find out the mean of the strings.


 




In [ ]:
df['Embarked'].fillna('S', inplace=True)      # NOW ALL THE NULL VALUES HAVE BEEN FILLED.
df.isnull().sum()

PassengerId    0
Survived       0
Pclass         0
Name           0
Sex            0
Age            0
SibSp          0
Parch          0
Ticket         0
Fare           0
Embarked       0
dtype: int64

**.corr() attribute tells of the about the correlation between different columns**

In [ ]:
df.corr()

PassengerId  Survived    Pclass  ...     SibSp     Parch      Fare
PassengerId     1.000000 -0.005007 -0.035144  ... -0.057527 -0.001652  0.012658
Survived       -0.005007  1.000000 -0.338481  ... -0.035322  0.081629  0.257307
Pclass         -0.035144 -0.338481  1.000000  ...  0.083081  0.018443 -0.549500
Age             0.033207 -0.069809 -0.331339  ... -0.232625 -0.179191  0.091566
SibSp          -0.057527 -0.035322  0.083081  ...  1.000000  0.414838  0.159651
Parch          -0.001652  0.081629  0.018443  ...  0.414838  1.000000  0.216225
Fare            0.012658  0.257307 -0.549500  ...  0.159651  0.216225  1.000000

[7 rows x 7 columns]





*   SibSp= Number of sibling/spouses aboard. 
* Parch=  Number of Parents/children aboard.
*   we can make a new column Familysize by combining these two columns.



In [ ]:
df['Familysize']=df['Parch']+df['SibSp']
df.drop(['SibSp','Parch'], axis=1, inplace=True)
df.corr()

PassengerId  Survived    Pclass       Age      Fare  Familysize
PassengerId     1.000000 -0.005007 -0.035144  0.033207  0.012658   -0.040143
Survived       -0.005007  1.000000 -0.338481 -0.069809  0.257307    0.016639
Pclass         -0.035144 -0.338481  1.000000 -0.331339 -0.549500    0.065997
Age             0.033207 -0.069809 -0.331339  1.000000  0.091566   -0.248512
Fare            0.012658  0.257307 -0.549500  0.091566  1.000000    0.217138
Familysize     -0.040143  0.016639  0.065997 -0.248512  0.217138    1.000000



*  **Familysize in the ship does not have much correlance with survival rate.**
*   let's see whether the person was alone or not  affect the survival rate.





In [ ]:
df['Alone']=[0 if df['Familysize'][i]>0 else 1 for i in df.index]
df.head()

PassengerId  Survived  Pclass  ... Embarked Familysize  Alone
0            1         0       3  ...        S          1      0
1            2         1       1  ...        C          1      0
2            3         1       3  ...        S          0      1
3            4         1       1  ...        S          1      0
4            5         0       3  ...        S          0      1

[5 rows x 11 columns]

In [ ]:
df.groupby(['Alone'])['Survived'].mean()

Alone
0    0.505650
1    0.303538
Name: Survived, dtype: float64



*   If the person is alone is he or she has less chances of surviving.


> The reason might be the person who is travelling with his family might be belonging to rich class and might be  prioritized over other.









In [ ]:
df[['Alone','Fare']].corr()

Alone      Fare
Alone  1.000000 -0.271832
Fare  -0.271832  1.000000



*   So we can see if the person was not alone the chance of the ticket  fare is higher is high.



In [ ]:
df['Sex'] = [0 if df['Sex'][i]=='male' else 1 for i in df.index]    # 1 for female, 0 for male.
df.groupby(['Sex'])['Survived'].mean()

Sex
0    0.188908
1    0.742038
Name: Survived, dtype: float64



*   It shows that  female passengers have more chance of surviving than  male ones. It shows  women were prioritized over men.




In [ ]:
df.groupby(['Embarked'])['Survived'].mean()

Embarked
C    0.553571
Q    0.389610
S    0.339009
Name: Survived, dtype: float64

In [ ]:
df.groupby(['Pclass'])['Survived'].mean()

Pclass
1    0.629630
2    0.472826
3    0.242363
Name: Survived, dtype: float64

# **CONCLUSION**: 







*   Female passengers are prioritized over men.
*   People with high class(rich people) have high chance of survival rate than others.

*   Passengers travelling with thier families have high rate of survival.
*   Passengers who borded  the ship at Cherbourg survived more in proportion to others.


*   Passengers who are in 1st class have more survival rate in proporthion to passengers in other class.It shows that people in rich class or first class are more prioritized than other.
*   There are more males in the ship than females.






